C++/include/data_structure/stack/stack.hpp

```diff
@@ -0,0 +1,81 @@
+/*
+    Stack
+    ------------------
+    A data structure which has the Last In First Out (LIFO) property. Elements are added
+    and removed from the top(head). Only the top element can be accessed at any given time.
+    Hence, there is no random access. All member functions have O(1) time complexity.
+*/
+#ifndef stack_h
+#define stack_h
+
+#include "data_structure/linked_list/singly_linked_list.hpp"
+#include <iostream>
```

Comment:
You shouldn't need to include `iostream` (remove the call to `std::cout` in `top()` and `pop()`). More discussion on this [here](https://stackoverflow.com/a/7390181/3292279), but basically I think it would be better to include an `isEmpty()` function for caller to test before calling `top()` or `pop()`. Make sure to add this to unit tests, too.